preview/Win7Msix/Win7MSIXInstaller/Extractor.cpp

```diff
@@ -276,9 +276,12 @@ HRESULT Extractor::CreateHandler(MsixRequest * msixRequest, IPackageHandler ** i
 
 HRESULT Extractor::ExtractPackage()
 {
-    RETURN_IF_FAILED(ExtractFootprintFiles());
-    RETURN_IF_FAILED(ExtractPayloadFiles());
-    RETURN_IF_FAILED(ExtractRegistry(false));
+    if(!m_msixRequest->GetIsCancelClicked())
+        RETURN_IF_FAILED(ExtractFootprintFiles());
+    if (!m_msixRequest->GetIsCancelClicked())
+        RETURN_IF_FAILED(ExtractPayloadFiles());

```

Comment:
>ExtractPayloadFiles [](start = 25, length = 19)

extractPayloadFiles should check between files as well, as that's generally going to be the most cancellable period of time.

Also, always use curly braces {} even for one line if statements, to avoid mistakes.

Reply:
Added check for cancel button within extractPayloadFiles as well. Added braces.